db/document.go

```diff
@@ -1494,8 +1495,7 @@ func (doc *Document) IsInConflict(ctx context.Context, db *DatabaseCollectionWit
 	}
 	// we need to check if local CV version was generated from a revID if so we need to perform conflict check on rev tree history
 	if doc.HLV.HasRevEncodedCV() {
-		_, _, isConflictErr := db.revTreeConflictCheck(ctx, putOpts.RevTreeHistory, doc, putOpts.NewDoc.Deleted)
-		if isConflictErr != nil {
+		if doc.RevConflict {
```

Comment:
We still need potentially need a check for revtree conflict here, take the scenario:

- Local doc has rev tree generated CV (before receiving the incoming doc over the replicator) 
- Incoming doc arrives and also has legacy generated CV, we get to function `PutExistingCurrentVersion` given local doc has CV we won't enter the code block on line https://sourcegraph.com/github.com/couchbase/sync_gateway/-/blob/db/crud.go?L1357 
- So when checking if documents are in conflict we need to use rev tree history to do so (given all legacy rev generated CV's have the source `Revision+Tree+Encoding` so the HLV's will never be in conflict) 

So essentially we want to keep `revTreeConflictCheck` here but only run if we haven't already done so I think.